### DISCUSSION TITLE
How to perform FUOTA during the event based transmission?

### DISCUSSION THREAD

**Discussion by shabaz-git** [Lora-net/LoRaMac-node#1212] [Q&A]
Hi all,

The default class of the node is Class-A and is powered with a continuous power supply.
By event-based uplink means, I will be checking the sensor data and whenever there's a significant change then only schedule an uplink.

So, in this scenario, the handshaking required for the FUOTA may not be possible when there are no uplinks and my device is not scheduling the uplinks as there is no new data.

Can maintaining the devices default class as Class-C do the work?

Please share any ideas?

**Comment by praneet-kaur** [comment]
Hi @shabaz-git ,
I came across this issue while performing clock sync for my end device. My temporary solution is to send few random uplinks whenever you want to perform FUOTA.
Another issue along the same line is, is there any recommended practice for regular uplinks while FUOTA is in process. Are we supposed to allow regular uplinks when any FUOTA event is going on or halt all uplinks other than FUOTA until FUOTA is complete. @mluis1 please advice. TIA.

**Reply by shabaz-git** [reply]
Hi @praneet-kaur ,

> send few random uplinks whenever you want to perform FUOTA

That is the limitation in my case. The node won't be aware of the FUOTA start as there is no way we can communicate with it in Class-A mode until and unless there's an uplink. To  100s of nodes. 
To establish the communication the node should be configured as Class-C. I did the changes with respect to Class-C and it's working fine.
Keeping the battery-powered node in Class-C will eat up the battery. In my case nodes are currently continuos powered. 
But need to think of another workaround to make the nodes work on the battery power, class-A with event-based uplinks and should support FUOTA also.

>  is there any recommended practice for regular uplinks while FUOTA is in process

I don't think scheduling uplinks during the FUOTA process is a good idea as there are chances of losing the firmware fragments.
